src/commands/delegate/swap.ts

```diff
@@ -0,0 +1,192 @@
+import chalk from 'chalk'
+import { Command } from '@oclif/command'
+import { ethers } from 'ethers'
+import * as utils from '../../lib/utils'
+import { getWallet } from '../../lib/wallet'
+import { get, cancelled } from '../../lib/prompt'
+const Delegate = require('@airswap/delegate/build/contracts/Delegate.sol/Delegate.json')
+const delegateDeploys = require('@airswap/delegate/deploys.js')
+const IERC20 = require('@airswap/utils/build/src/abis/ERC20.json')
+
+export default class DelegateSetRule extends Command {
+  public static description = 'set a delegate rule'
+
+  public async run() {
+    try {
+      const wallet = await getWallet(this, true)
+      console.log(wallet.address)
+      const chainId = (await wallet.provider.getNetwork()).chainId
+      utils.displayDescription(this, DelegateSetRule.description, chainId)
+
+      const delegateContract = new ethers.Contract(
+        delegateDeploys[chainId],
+        Delegate.abi,
+        wallet
+      )
+
+      this.log(chalk.white(`Delegate contract: ${delegateContract.address}\n`))
+
+      const { senderWallet }: any = await get({
```

Comment:
Rather than punch these parameters in, we could prompt for: senderAmount ("amount"), senderToken ("of"), senderToken ("for"), and either prompt or scan logs for senderWallet. Then create and sign an order (see `createOrderERC20` in stream.ts) and submit a transaction to the Delegate